db-engine/src/main/java/com/splicemachine/db/iapi/sql/execute/ResultSetFactory.java

```diff
@@ -2105,60 +2105,62 @@ NoPutResultSet getRowCountResultSet(
+                                      int resultSetNumber,
+                                      String exportPath,
+                                      String compression,
+                                      int replicationCount,
+                                      String encoding,
+                                      String fieldSeparator,
+                                      String quoteChar,
+                                      int srcResultDescriptionSavedObjectNum) throws StandardException;
+
+
+    /**
+     * Binary Export
+     */
+    NoPutResultSet getBinaryExportResultSet(NoPutResultSet source,
+                                      Activation activation,
+                                      int resultSetNumber,
+                                      String exportPath,
+                                      String compression,
+                                      String format,
+                                      int srcResultDescriptionSavedObjectNum) throws StandardException;
     /**
      * Batch Once
      */
-	NoPutResultSet getBatchOnceResultSet(NoPutResultSet source,
-										 Activation activation,
-										 int resultSetNumber,
-										 NoPutResultSet subqueryResultSet,
-										 String updateResultSetFieldName,
-										 int sourceCorrelatedColumnItem,
-										 int subqueryCorrelatedColumnItem) throws StandardException;
-
-	/**
-	 * Recursive query
-	 */
-	NoPutResultSet getSelfReferenceResultSet(Activation activation,
-											 GeneratedMethod rowAllocator,
-											 int resultSetNumber,
-											 double optimizerEstimatedRowCount,
-											 double optimizerEstimatedCost,
-											 String explainPlan) throws StandardException;
-
-	NoPutResultSet getRecursiveUnionResultSet(NoPutResultSet leftResultSet,
-											  NoPutResultSet rightResultSet,
-											  int resultSetNumber,
-											  double optimizerEstimatedRowCount,
-											  double optimizerEstimatedCost,
-											  String explainPlan,
-											  int iterationLimit) throws StandardException;
-
-	NoPutResultSet getSignalResultSet(Activation activation,
-	                                  String sqlState,
-	                                  GeneratedMethod errorTextGenerator) throws StandardException;
+    NoPutResultSet getBatchOnceResultSet(NoPutResultSet source,
+                                         Activation activation,
+                                         int resultSetNumber,
+                                         NoPutResultSet subqueryResultSet,
+                                         String updateResultSetFieldName,
+                                         int sourceCorrelatedColumnItem,
+                                         int subqueryCorrelatedColumnItem,
+                                         int sourceRowLoationColumnPosition,
+                                         int cardinalityCheck) throws StandardException;
```

Comment:
Are we forcing the recompilation of stored prepared statements? Otherwise we might break some with this change (triggers for instance)

Reply:
Thanks Daniel! Yes, we need to force a re=compilation of stored prepared statement. I've added an upgrade script to do that and tested upgrade with a trigger case.